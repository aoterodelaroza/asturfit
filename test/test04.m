#! /usr/bin/octave -q
% Copyright (C) 2010 Victor Lua~na and Alberto Otero-de-la-Roza
%
% This octave routine is free software: you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation, either version 3 of the License, or (at
% your option) any later version. See <http://www.gnu.org/licenses/>.
%
% The routine distributed in the hope that it will be useful, but WITHOUT
% ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
% FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
% more details.

% Compare the properties of bcc Li to Rb.

global nelectrons

addpath("../src/");

# elements
element = {"Na", "K", "Rb"};

hybohr3togpa = 2*14710.50498740275538944426;
fac = [1,1,hybohr3togpa,1,1/hybohr3togpa,1/hybohr3togpa^2];
 
# Table: equilibrium properties of the elements
printf("El");
printf(" -Eceil-");
printf(" --V0-(bohr^3)---");
printf(" --E0-(hartree)--");
printf(" ----B0-(GPa)----");
printf(" ------B1p0------");
printf(" --B2p0-(1/GPa)--");
printf("\n");
for i = 1 : length(element)
   el = tolower(element{i});
   filein = sprintf("w2k-lda-%s.dat", el);
   [v{i},e{i}] = readEVdata(filein,1);
   vref{i} = median(v{i});
   eceil{i} = ceil(max(e{i}));
   e{i} -= eceil{i};
   [cf{i},sf{i}] = avgstrainfit(v{i},e{i},vref{i},6,:,:,0);
   printf("%-2s", element{i});
   printf(" %7d", eceil{i});
   for k = 1 : 5
      printf(" %-16s", errformatf(sf{i}.eqmean(k)*fac(k), sf{i}.eqstd(k)*fac(k)));
   endfor
   printf("\n");
   rk = find(sf{i}.pmean > 0);
   vpos{i} = v{i}(rk);
endfor

# Figures will not appear on screen:
set(gcf(),"visible","off");

# Figure: E(V) for all the elements.
# We will make sure that all elements have the same number of points.
V = E = [];
for i = 1 : length(element)
   vv = linspace(min(v{i}), max(v{i}), 201)';
   ee = strainevalE(cf{i}, vref{i}, vv);
   V = [V, vv/sf{i}.eqmean(1)];
   E = [E, (ee-sf{i}.eqmean(2))/(sf{i}.eqmean(1)*sf{i}.eqmean(3))];
   key{i} = sprintf("-;%s;", element{i});
endfor
fileplot = "w2k-lda-ev.eps";
plot(V,E,key);
xlabel('V/V_0'); ylabel('(E-E_0)/(B_0V_0)'); grid('on');
print(fileplot, '-depsc');
printf("E(V/V_0) plot created in: %s\n", fileplot);

# Figures: B(p), B(V), B1p(p), B*B2p(p)
V = P = B = B1 = BB2 = [];
npts = 201; rk = 1:npts;
for i = 1 : length(element)
   vv = linspace(min(vpos{i}), max(vpos{i}), npts)';
   vv = vv(rk);
   s = straineval(cf{i}, vref{i}, vv);
   V = [V, vv/sf{i}.eqmean(1)];
   P = [P, s.p*hybohr3togpa];
   B = [B, s.B*hybohr3togpa];
   B1 = [B1, s.B1p];
   BB2 = [BB2, s.B.*s.B2p];
   key{i} = sprintf("-;%s;", element{i});
endfor

fileplot = "w2k-lda-bp.eps";
plot(P,B,key);
xlabel('p_{static} (GPa)'); ylabel('B (GPa)'); grid('on');
print(fileplot, '-depsc');
printf("B(p) plot created in: %s\n", fileplot);

fileplot = "w2k-lda-bv.eps";
plot(V,B,key);
xlabel('V/V_0'); ylabel('B (GPa)'); grid('on');
print(fileplot, '-depsc');
printf("B(V/V_0) plot created in: %s\n", fileplot);

fileplot = "w2k-lda-b1p.eps";
plot(P,B1,key);
xlabel('p_{static} (GPa)'); ylabel("B^'"); grid('on');
print(fileplot, '-depsc');
printf("B1p(p) plot created in: %s\n", fileplot);

fileplot = "w2k-lda-b2pbp.eps";
plot(P,BB2,key);
xlabel('p_{static} (GPa)'); ylabel('B^{''}B'); grid('on');
print(fileplot, '-depsc');
printf("B2pB(p) plot created in: %s\n", fileplot);
