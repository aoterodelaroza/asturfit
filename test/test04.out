warning: mark_as_command is obsolete and will be removed from a future version of Octave
El -Eceil- --V0-(bohr^3)--- --E0-(hartree)-- ----B0-(GPa)---- ------B1p0------ --B2p0-(1/GPa)--
readEVdata: Reading input file -> w2k-lda-na.dat
Input volume unit: bohr^3
Input energy unit: ry
* Energy and volume ranges 
  V-range (bohr^3) : [159.9086000000 , 319.8173000000] dV = 159.9087000000
  E-range (hartree) : [-161.7088550000 , -161.7039449000] dE = 0.0049101000
  Number of data points: 61
  Number of electrons: 11

Na    -161 224.4521(42)     -0.708856(1)     9.169(11)        3.7361(34)       -0.369(14)      
readEVdata: Reading input file -> w2k-lda-k.dat
Input volume unit: bohr^3
Input energy unit: ry
* Energy and volume ranges 
  V-range (bohr^3) : [243.6753000000 , 728.0767000000] dV = 484.4014000000
  E-range (hartree) : [-600.6042370500 , -600.5888263500] dE = 0.0154107000
  Number of data points: 100
  Number of electrons: 19

K     -600 432.2(11)        -0.604191(15)    4.485(22)        3.826(90)        -0.871(63)      
readEVdata: Reading input file -> w2k-lda-rb.dat
Input volume unit: bohr^3
Input energy unit: ry
* Energy and volume ranges 
  V-range (bohr^3) : [188.1585000000 , 1004.8819000000] dV = 816.7234000000
  E-range (hartree) : [-2977.9368343500 , -2977.8858850500] dE = 0.0509493000
  Number of data points: 101
  Number of electrons: 37

Rb   -2977 523.9(17)        -0.936835(14)    3.573(15)        3.721(82)        -1.190(59)      
E(V/V_0) plot created in: w2k-lda-ev.eps
B(p) plot created in: w2k-lda-bp.eps
B(V/V_0) plot created in: w2k-lda-bv.eps
B1p(p) plot created in: w2k-lda-b1p.eps
B2pB(p) plot created in: w2k-lda-b2pbp.eps
